### PR TITLE
UI Test : Introduce security for prototype APIs

### PR DESCRIPTION
**Pull request by RusJaI**
Test suite covers the three possible flows 

  1. user can successfully try out the prototype APIs at dev portal when security enabled and token not provided
  2. user is not allowed to try out the prototype APIs at dev portals when security disabled but token not provided (401 unauthorized response)
  3. user can successfully try out the prototype APIs at dev portal when security enabled and token provided

### CODE REVIEW

**Review by chanaka3d — APPROVED** [review]

File: portals/publisher/source/src/app/components/Apis/Listing/components/ImageGenerator/APICards/ApiThumbClassic.jsx [hunk 1]
```diff
@@ -215,6 +215,7 @@ class APIThumb extends Component {
                 onBlur={this.toggleMouseOver}
                 elevation={isHover ? 4 : 1}
                 className={classes.card}
+                data-testid={'card-'+api.name+api.version}
```
Comment (tharikaGitHub) [inline portals/publisher/source/src/app/components/Apis/Listing/components/ImageGenerator/APICards/ApiThumbClassic.jsx:218]: Let's fix the formatting here

File: portals/publisher/source/src/app/components/Apis/Listing/components/ImageGenerator/APICards/ApiThumbClassic.jsx [hunk 2]
```diff
@@ -285,7 +286,7 @@ class APIThumb extends Component {
                         </div>
                     </div>
                 </CardContent>
-                <CardActions className={classes.apiActions}>
+                <CardActions className={classes.apiActions} data-testid={'card-action-'+api.name+api.version}>
```
Comment (tharikaGitHub) [inline portals/publisher/source/src/app/components/Apis/Listing/components/ImageGenerator/APICards/ApiThumbClassic.jsx:289]: Formatting

File: tests/cypress/support/commands.js [hunk 3]
```diff
@@ -134,12 +134,15 @@ Cypress.Commands.add('deleteUser', (name) => {
 });
 
 Cypress.Commands.add('deleteApi', (name, version) => {
+    var cardName='card-'+name+version;
```
Comment (tharikaGitHub) [inline tests/cypress/support/commands.js:137]: Formatting

File: tests/cypress/support/commands.js [hunk 4]
```diff
@@ -739,10 +749,11 @@ Cypress.Commands.add('createAndPublishApi', (apiName = null) => {
 })
 
 Cypress.Commands.add('logoutFromDevportal', (referer = '/devportal/apis') => {
-    cy.visit(`${Utils.getAppOrigin()}/devportal/apis?tenant=carbon.super`);
-    cy.wait(2000);
+    //cy.visit(`${Utils.getAppOrigin()}/devportal/apis?tenant=carbon.super`);
```
Comment (tharikaGitHub) [inline tests/cypress/support/commands.js:752]: Let's remove commented code

File: portals/publisher/source/src/app/components/Apis/Details/Environments/Environments.jsx [hunk 5]
```diff
@@ -1635,6 +1635,7 @@ export default function Environments() {
                                                                 icon={<RadioButtonUncheckedIcon />}
                                                                 checkedIcon={<CheckCircleIcon color='primary' />}
                                                                 inputProps={{ 'aria-label': 'secondary checkbox' }}
+                                                                data-testid={row.displayName+'gateway-select-btn'}
```
Comment (hisanhunais) [inline portals/publisher/source/src/app/components/Apis/Details/Environments/Environments.jsx:1638]: Let's fix the formatting here.

File: tests/cypress/integration/publisher/020-prototyped-apis/00-try-out-resources-without-key-when-security-disabled.spec.js [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+describe("prototype apis with security disabled", () => {
+    const userName = 'admin';
+    const password = 'admin';
+    const apiName="Prototyped_sample";
+    const apiVersion='1.0.0';
```
Comment (hisanhunais) [inline tests/cypress/integration/publisher/020-prototyped-apis/00-try-out-resources-without-key-when-security-disabled.spec.js:23]: Let's fix the formatting here.

File: tests/cypress/integration/publisher/020-prototyped-apis/01-cannot-try-out-resources-without-key-when-security-enabled.spec.js [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+describe("prototype apis with security enabled", () => {
+    const userName = 'admin';
+    const password = 'admin';
+    const apiName="Prototyped_sample";
+    const apiVersion='1.0.0';
```
Comment (hisanhunais) [inline tests/cypress/integration/publisher/020-prototyped-apis/01-cannot-try-out-resources-without-key-when-security-enabled.spec.js:23]: Let's fix the formatting here.

File: tests/cypress/integration/publisher/020-prototyped-apis/02-can-try-out-resources-with-key-when-security-enabled.spec.js [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+describe("prototype apis with security enabled", () => {
+    const userName = 'admin';
+    const password = 'admin';
+    const apiName="Prototyped_sample";
+    const applicationName="Prototype client app";
+    const apiVersion='1.0.0';
```
Comment (hisanhunais) [inline tests/cypress/integration/publisher/020-prototyped-apis/02-can-try-out-resources-with-key-when-security-enabled.spec.js:24]: Let's fix the formatting here.

File: tests/cypress/integration/publisher/020-prototyped-apis/02-can-try-out-resources-with-key-when-security-enabled.spec.js [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+
+        cy.get("#left-menu-itemlifecycle").click();
+        cy.get('[data-testid="Deploy as a Prototype-btn"]',{timeout:3000}).click();
+
+        cy.logoutFromPublisher();
+
+        //login to dev portal as Developer
+        cy.loginToDevportal(userName, password);
+        
+        cy.createApplication(applicationName,"50PerMin","Sample Description");
```
Comment (hisanhunais) [inline tests/cypress/integration/publisher/020-prototyped-apis/02-can-try-out-resources-with-key-when-security-enabled.spec.js:72]: Let's fix the formatting here.

File: tests/cypress/support/commands.js [hunk 10]
```diff
@@ -649,27 +652,34 @@ Cypress.Commands.add('addProperty',(name,value,ifSendToDevPortal)=>{
 
 })
 
-Cypress.Commands.add('createAPIWithoutEndpoint', (name, type = 'REST') => {
+Cypress.Commands.add('createAPIWithoutEndpoint', (name=null,version=null,type = 'REST') => {
```
Comment (hisanhunais) [inline tests/cypress/support/commands.js:655]: Let's fix the formatting here.

File: tests/cypress/support/commands.js [hunk 11]
```diff
@@ -649,27 +652,34 @@ Cypress.Commands.add('addProperty',(name,value,ifSendToDevPortal)=>{
...
     const random_number = Math.floor(Date.now() / 1000);
-    const randomName = `0sample_api_${random_number}`;
-    cy.visit(`${Utils.getAppOrigin()}/publisher/apis`)
-    cy.get('#itest-rest-api-create-menu', { timeout: 30000 });
-    cy.get('#itest-rest-api-create-menu').click();
+    var apiVersion=`v${random_number}`;
```
Comment (hisanhunais) [inline tests/cypress/support/commands.js:657]: Let's fix the formatting here.

File: tests/cypress/support/commands.js (hunk 3 above)
Comment (RusJaI) [reply tests/cypress/support/commands.js:137]: fixed

File: tests/cypress/support/commands.js (hunk 4 above)
Comment (RusJaI) [reply tests/cypress/support/commands.js:752]: this has already been fixed by a later commit

File: portals/publisher/source/src/app/components/Apis/Listing/components/ImageGenerator/APICards/ApiThumbClassic.jsx (hunk 1 above)
Comment (RusJaI) [reply portals/publisher/source/src/app/components/Apis/Listing/components/ImageGenerator/APICards/ApiThumbClassic.jsx:218]: fixed

File: portals/publisher/source/src/app/components/Apis/Listing/components/ImageGenerator/APICards/ApiThumbClassic.jsx (hunk 2 above)
Comment (RusJaI) [reply portals/publisher/source/src/app/components/Apis/Listing/components/ImageGenerator/APICards/ApiThumbClassic.jsx:289]: fixed

File: portals/publisher/source/src/app/components/Apis/Details/Environments/Environments.jsx (hunk 5 above)
Comment (RusJaI) [reply portals/publisher/source/src/app/components/Apis/Details/Environments/Environments.jsx:1638]: fixed

File: tests/cypress/integration/publisher/020-prototyped-apis/00-try-out-resources-without-key-when-security-disabled.spec.js (hunk 6 above)
Comment (RusJaI) [reply tests/cypress/integration/publisher/020-prototyped-apis/00-try-out-resources-without-key-when-security-disabled.spec.js:23]: fixed

File: tests/cypress/integration/publisher/020-prototyped-apis/01-cannot-try-out-resources-without-key-when-security-enabled.spec.js (hunk 7 above)
Comment (RusJaI) [reply tests/cypress/integration/publisher/020-prototyped-apis/01-cannot-try-out-resources-without-key-when-security-enabled.spec.js:23]: fixed

File: tests/cypress/integration/publisher/020-prototyped-apis/02-can-try-out-resources-with-key-when-security-enabled.spec.js (hunk 8 above)
Comment (RusJaI) [reply tests/cypress/integration/publisher/020-prototyped-apis/02-can-try-out-resources-with-key-when-security-enabled.spec.js:24]: fixed

File: tests/cypress/integration/publisher/020-prototyped-apis/02-can-try-out-resources-with-key-when-security-enabled.spec.js (hunk 9 above)
Comment (RusJaI) [reply tests/cypress/integration/publisher/020-prototyped-apis/02-can-try-out-resources-with-key-when-security-enabled.spec.js:72]: fixed

File: tests/cypress/support/commands.js (hunk 10 above)
Comment (RusJaI) [reply tests/cypress/support/commands.js:655]: fixed

File: tests/cypress/support/commands.js (hunk 11 above)
Comment (RusJaI) [reply tests/cypress/support/commands.js:657]: fixed

**Pull request merged**